src/main/java/ch/njol/skript/expressions/ExprJoinSplit.java

```diff
@@ -40,65 +41,72 @@
  */
 @Name("Join & Split")
 @Description("Joins several texts with a common delimiter (e.g. \", \"), or splits a text into multiple texts at a given delimiter.")
-@Examples({"message \"Online players: %join all players with \"\" | \"\"%\" # %all players% would use the default \"x, y, and z\"",
-		"set {_s::*} to the string argument split at \",\""})
-@Since("2.1, 2.5.2 (regex support)")
+@Examples({
+		"message \"Online players: %join all players with \"\" | \"\"%\" # %all players% would use the default \"x, y, and z\"",
+		"set {_s::*} to the string argument split at \",\""
+})
+@Since("2.1, 2.5.2 (regex support), INSERT VERSION (case sensitivity)")
 public class ExprJoinSplit extends SimpleExpression<String> {
-	
+
 	static {
 		Skript.registerExpression(ExprJoinSplit.class, String.class, ExpressionType.COMBINED,
 			"(concat[enate]|join) %strings% [(with|using|by) [[the] delimiter] %-string%]",
-			"split %string% (at|using|by) [[the] delimiter] %string%",
-			"%string% split (at|using|by) [[the] delimiter] %string%",
+			"split %string% (at|using|by) [[the] delimiter] %string% [case:with case sensitivity]",
+			"%string% split (at|using|by) [[the] delimiter] %string% [case:with case sensitivity]",
 			"regex split %string% (at|using|by) [[the] delimiter] %string%",
 			"regex %string% split (at|using|by) [[the] delimiter] %string%");
 	}
-	
+
 	private boolean join;
 	private boolean regex;
-	
+	private boolean caseSensitivity;
+
 	@SuppressWarnings("null")
 	private Expression<String> strings;
 	@Nullable
 	private Expression<String> delimiter;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings({"unchecked", "null"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		join = matchedPattern == 0;
 		regex = matchedPattern >= 3;
+		caseSensitivity = SkriptConfig.caseSensitive.value() || parseResult.hasTag("case");
 		strings = (Expression<String>) exprs[0];
 		delimiter = (Expression<String>) exprs[1];
 		return true;
 	}
-	
+
 	@Override
 	@Nullable
-	protected String[] get(final Event e) {
-		final String[] s = strings.getArray(e);
-		final String d = delimiter != null ? delimiter.getSingle(e) : "";
+	protected String[] get(Event event) {
+		String[] s = strings.getArray(event);
+		String d = delimiter != null ? delimiter.getSingle(event) : "";
```

Comment:
In Java you can actually use `this` to reference the field outside the code block to keep the variable naming.
```suggestion
		String[] strings = this.strings.getArray(event);
		String delimiter = this.delimiter != null ? this.delimiter.getSingle(event) : "";
```
*Other changes requires under this suggestion block*